File: Leetcode/404. Sum of Left Leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (root == null) return 0;
        int Sum = 0;
        if (root.left != null) {
            if (root.left.left == null && root.left.right == null) Sum += root.left.val;
        }
        Sum += sumOfLeftLeaves (root.left) + sumOfLeftLeaves (root.right);
        return Sum;
    }
}

File: Leetcode/Trees/404. Sum of Left Leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        return 0 if not root else self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right) + (root.left.val if root.left and not root.left.left and not root.left.right else 0)